tests/cpptests/test_cpp_thpool.cpp

```diff
@@ -207,3 +207,84 @@ TEST_P(PriorityThpoolTestFunctionality, TestPauseResume) {
     ASSERT_EQ(redisearch_thpool_num_jobs_in_progress(this->pool), 0) << "expected 0 working threads";
 
 }
+
+/* ========================== NUM_THREADS = 2, NUM_PRIVILEGED = 1 ========================== */
+THPOOL_TEST_SUITE(PriorityThpoolTestPrivilegedAndUnprivileged, 2, 1)
+
+TEST_P(PriorityThpoolTestPrivilegedAndUnprivileged, TestTakingAsPrivileged) {
+    size_t num_jobs = 5;
+
+    // This job will keep the thread busy until we tell it to finish.
+    auto waitForSignFunc = [](void *p) {
+        bool *waitForSign = (bool *)p;
+        while (*waitForSign) {
+            usleep(1);
+        }
+    };
+    // This job will signal the waiting job to finish.
+    auto signalFunc = [](void *p) {
+        bool *waitForSign = (bool *)p;
+        *waitForSign = false;
+    };
+    // This job will count how many times it was taken with a specific priority.
+    auto countFunc = [](void *p) {
+        size_t *count = (size_t *)p;
+        (*count)++;
+    };
+    // This job will check the state of the counts.
+    struct state_test {
+        size_t *countHigh;
+        size_t *countLow;
+
+        size_t expectedHigh;
+        size_t expectedLow;
+    };
+    auto stateCheck = [](void *p) {
+        state_test *state = (state_test *)p;
+        ASSERT_EQ(*state->countHigh, state->expectedHigh);
+        ASSERT_EQ(*state->countLow, state->expectedLow);
+    };
+
+    /* Scenario of the test:
+     * 1. We add 2 waiting jobs with low priority, so both threads will be unprivileged, and we can control
+     *    which thread will take the next job.
+     * 2. We add 5 count jobs with high priority and 5 count jobs with low priority.
+     * 3. We add a state check job for each priority.
+     * 4. Add a final job to release the other waiting thread (low priority job)
+     * 5. Release one of the threads from the waiting job to execute the rest of the jobs.
+     *
+     * We expect that the unblocked thread will understand that it is privileged and will take the high priority jobs
+     * before the low priority jobs. therefore, the high priority state check is expected to see 5 high priority jobs
+     * and 0 low priority jobs, and the low priority state check is expected to see 5 high priority jobs and 5 low priority jobs.
+     *
+     * Before the mechanism fix, the unblocked thread would see that there is one running thread, and would assume that it is
+     * unprivileged (num jobs in progress is not smaller than the number of privileged threads), and would take once a high priority
```

Comment:
wasn't clear to me if "it" refers to the blocked or to the unblocked thread. 
Feel free to rephrase 
```suggestion
     * Before the mechanism fix, the unblocked thread would assume that it is unprivileged if it sees that there is one running thread
     * (num jobs in progress is not smaller than the number of privileged threads), and would take once a high priority
```